Freeze DAML-LF 1.8

cocreature opened the pull request:
Two minor points that I did not mention in the previous PR:

We also include the renaming of structural records to `struct` and the
renaming of `Map` to `TextMap`.

There are some minor changes around the LF encoder tests which need to
be able to emit package metadata properly so I’ve added it to the
parser. Sorry for not splitting that out.

Following the process used for the DAML-LF 1.7 release, this does not
yet include the frozen proto file.

changelog_begin

- [DAML-LF] Release DAML-LF 1.8:

  * Rename structural records to ``Struct``. Note that
    structural records are not exposed in DAML.
  * Rename ``Map`` to ``TextMap``.
  * Add type synonyms. Note that type synonyms are not serializable.
  * Add package metadata, i.e., package names and versions.

  Note that the default output of ``damlc`` is stil DAML-LF 1.7. You
  can produce DAML-LF 1.8 by passing ``--target=1.8``.

changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

cocreature commented on daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/VersionTimeline.scala:
```diff
@@ -56,9 +56,10 @@ object VersionTimeline {
       Both(This(ValueVersion("5")), LanguageVersion(LMV.V1, "6")),
       Both(This(ValueVersion("6")), LanguageVersion(LMV.V1, "7")),
       This(That(TransactionVersion("9"))),
+      That(LanguageVersion(LMV.V1, "8")),
```
@remyhaemmerle-da I don’t really understand this file. Based on the TODO, I believe that this should go before `ValueVersion("7")` since we are not including `GenMap` in DAML-LF 1.8. But please check that this matches the intent.

remyhaemmerle-da replied:
Please delete the `FIXME` with the `This(This(ValueVersion("7"))),`

cocreature replied:
Do you want me to delete just the FIXME or also the `This(This(ValueVersion("7"))` line?

remyhaemmerle-da replied:
Yep

remyhaemmerle-da replied:
You did good.

remyhaemmerle-da commented on daml-lf/encoder/BUILD.bazel:
```diff
@@ -21,6 +21,7 @@ lf_targets = [
     "1.3",
     "1.6",
     "1.7",
+    "1.8",
```
This require also update of file names in  `daml-lf/encoder/src/test/lf` 
(I know this is ugly)

remyhaemmerle-da replied:
I also expect a test for synonyms here.

cocreature replied:
I’ve updated this and added a test for type synonyms which works (after actually implementing the encoder).

remyhaemmerle-da replied:
:+1:

cocreature commented:
>     * `daml/daml-lf/language/daml-lf.bzl`

Unless I’m missing something this is already updated. `lf_stable_version` is still set to `1.6`. I don’t know what that is supposed to represent and given that it wasn’t 1.7 so far, it seems unrelated to this change.

remyhaemmerle-da commented:
> > ```
> > * `daml/daml-lf/language/daml-lf.bzl`
> > ```
> 
> Unless I’m missing something this is already updated. `lf_stable_version` is still set to `1.6`. I don’t know what that is supposed to represent and given that it wasn’t 1.7 so far, it seems unrelated to this change.

`lf_stable_version` is typically the default output of the compiler.
`lf_latest_version` is the latest supported version by the engine (this can be a version not supported by the compiler)

cocreature commented:
> `lf_stable_version` is typically the default output of the compiler.
> `lf_latest_version` is the latest supported version by the engine (this can be a version not supported by the compiler).

I see, can we bump this in a separate PR? I expect that there will be a bunch of test failures if we bump this to 1.7 now and those are unrelated to this PR so I’d like to separate them.

cocreature commented:
> * ``
> 
> * `language-support/java/codegen/BUILD.bazel`
> 
> * `language-support/scala/codegen-sample-app/BUILD.bazel` (the comment in line 24)

I’ve updated the tests for the Java codegen. For the Scala codegen, I’ve just updated the comment. Afaict it uses `1.dev` for `GenMap` so it has to keep using it.

remyhaemmerle-da commented:
> > `lf_stable_version` is typically the default output of the compiler.
> > `lf_latest_version` is the latest supported version by the engine (this can be a version not supported by the compiler).
> 
> I see, can we bump this in a separate PR? I expect that there will be a bunch of test failures if we bump this to 1.7 now and those are unrelated to this PR so I’d like to separate them.

If fact I mistake, it is the latest version the compiler support. 
I do not expect failure if you update this value. 

If you make sure their is no release the SDK, before merging this separate PR, it is fine with me.

cocreature commented:
I’ve bumped `lf_stable_version` to `1.7`. I believe I’ve addressed all comments now.
